### PR TITLE
Rename "Associated*" to "Assoc*"

### PR DESCRIPTION
**Pull request by agnxy**
This change is for #60163.

r? @oli-obk

### CODE REVIEW

File: src/libsyntax/ast.rs [hunk 1]
```diff
@@ -1253,11 +1253,11 @@ pub enum ExprKind {
 /// item qualified with this `Self` type.
 ///
 /// ```ignore (only-for-syntax-highlight)
-/// <Vec<T> as a::b::Trait>::AssociatedItem
+/// <Vec<T> as a::b::Trait>::AssocItem
```
Comment (oli-obk) [inline src/libsyntax/ast.rs]: these aren't internal datastructures, please keep the original here

File: src/librustc/error_codes.rs [hunk 2]
```diff
@@ -856,23 +856,23 @@ Examples follow.
 Here is a basic example:
 
 ```compile_fail,E0271
-trait Trait { type AssociatedType; }
+trait Trait { type AssocType; }
```
Comment (oli-obk) [inline src/librustc/error_codes.rs]: these aren't internal datastructures, please keep the original here

**Review by oli-obk — COMMENTED** [review]
lgtm, just some nits and needs a rebase

File: src/libsyntax/ast.rs (hunk 1 above)
Comment (agnxy) [reply src/libsyntax/ast.rs]: Thanks for your review. It's fixed.

**Comment by oli-obk** [comment]
Thanks, this is ready to merge now, but we need to wait until the beta is cut in the next few days as we can't break tools right now

**Comment by oli-obk** [comment]
If you want to be proactive about the breakage, you can run `./x.py test src/tools/{clippy,rls,miri}` and fix all breakage and open PRs against the tools.

**Comment by agnxy** [comment]
Thanks @oli-obk . I'm not aware this is a breaking change😅 I'll take a look at [CONTRIBUTING.md](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) first.

**Comment by Centril** [comment]
Failed in https://github.com/rust-lang/rust/pull/61198#issuecomment-495955551. @bors r-

**Comment by agnxy** [comment]
@Centril I have fixed the build failure at the tip of master branch.

**Pull request merged**